tools/c7n_mailer/c7n_mailer/gcp/gcp_pubsub_processor.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import six
+import json
+from c7n_mailer.email_delivery import EmailDelivery
+from c7n_gcp.client import Session
+import base64
+import zlib
+
+MAX_MESSAGES = 1000
+
+
+class MailerGcpPubSubProcessor(object):
+
+    def __init__(self, config, logger, session=None):
+        self.config = config
+        self.logger = logger
+        self.subscription = self.config['queue_url']
+        self.session = session or Session()
```

Comment:
set up the client here as it's reused elsewhere in the other functions

Reply:
Fixed in next commit.